savoirfairelinux/sous-chef
# New: member.Relationship model, combining referents and emergency contacts. Closes #740

## Fixes #740  by lingxiaoyang

### Changes proposed in this pull request:

* Model change: adding `member.Relationship`, deleting `member.Referencing` and `member.EmergencyContact`. Migration `member.0032` is provided to take care of existing database.
* Forms: creating relationship formsets based on emergency contact formsets. Deleting referent form and emergency contact formsets.
* Views and templates: removing referent and emergency contacts from client form wizard. Inserting relationships after address information and before payment information in the wizard.
* Admin: replacing referent and emergency contacts with relationships.
* UI: adding dynamic fields in relationship form to support the extra fields of a referent relationship.
* Unit test: replacing referent and emergency contacts with relationships. Testing the forward and reverse migration of `member.0032`. Fixing a problem in migration test case that causes errors when there are multiple migration unit tests.
* Sample data fixture: replacing referent and emergency contacts with relationships.

### Status


### How to verify this change

1. `python3 manage.py migrate`. This should work fine and should transfer all old data into new relationship model.
2. Go to Admin -> Relationships to verify transferred relationship entries. They are also available in Client detail view as inline formsets.
3. Go to client information view to check the display of relationship entries.
4. Try editing the relationships of an existing client.
5. Try creating the relationships of a new client.

### Deployment notes and migration


### New translatable strings


### Additional notes

Needs more manual tests.
